enacts/flex_fcst/cpt.py

```diff
@@ -1,162 +1,228 @@
     ----------
-    data_path : str
-        String of the path pointing to cpt datasets.
-    filename_pattern : str
-        String of the filename pattern name for a given variable's data file.
-    regex_search_pattern : str
-        String representing regular expression search pattern to find dates in file names.
-    format_in : str
-        String representing dates format found in file names
-    format_out : str
-        String representing desired output dates format.
+    fcst_conf: dict
+        dictionary indicating pyCPT datasets configuration (see config)
+
     Returns
     -------
-    start_dates : list
-        List of strings representing all start dates for the data within `data_path`.
-    Notes
-    -----
-    For more information on regex visit: https://docs.python.org/3/library/re.html
-    Test your regex code here: https://regexr.com/
-    Examples
+    fcst_ds, obs : xr.Dataset, xr.DataArray
+        dataset of forecast mean and variance and their obs
+
+    See Also
     --------
-    Regex expression "\w{3}-\w{1,2}-\w{4}" matches expressions that are:
-    '{word of 3 chars}-{word between 1,2 chars}-{word of 4 chars}'
-    will match dates of format 'Apr-4-2022', 'dec-14-2022', etc.
+    read_pycptv2dataset
     """
-    filename_pattern = filename_pattern.replace("SLtarget", "*")
-    files_name_list = glob.glob(f'{data_path}/{filename_pattern}')
-    start_dates = []
-    for file in files_name_list:
-        start_date = re.search(regex_search_pattern, file)
-        start_date_dt = datetime.strptime(start_date.group(), format_in)
-        start_dates.append(start_date_dt)
-    start_dates = sorted(set(start_dates)) #finds unique dates in the case there are files with the same date due to multiple lead times
-    start_dates = [i.strftime(format_out) for i in start_dates]
-    return start_dates
+    if "forecast_path" in fcst_conf :
+        fcst_ds, obs = read_pycptv2dataset(fcst_conf["forecast_path"])
+    else :
+        raise Exception("No synthetic case as of yet")
+    return fcst_ds, obs
 
 
 def read_pycptv2dataset(data_path):
+    """ Create a xr.Dataset from yearly pyCPT nc files for one or more targets of the
+    year, for a forecast mean and variance variables, appending multiple Starts; and
+    the corresponding xr.DataArray of the observations time series used to train
+    those targets
+
+    Parameters
+    ----------
+    data_path: str
+        path of set of nc files, organized by targets if multiple ones.
+
+    Returns
+    -------
+    fcst_ds, obs : xr.Dataset, xr.DataArray
+        dataset of forecast mean and variance from `data_path` and their obs
+
+    See Also
+    --------
+    read_pycptv2dataset_single_target
+
+    Notes
+    -----
+    Whether `data_path` contains multiple targets or not is determined by whether the
+    observation obs.nc file(s) is a direct child of `data_path` or under target-wise
+    folders that have no restriction on how they are named.
```

Comment:
Seems like their names should be predetermined, not arbitrary, so we can just open the one we need instead of opening all of them and then picking the one we need (see my comment at the top of this review). Just a thought for another day.